src/csi-wrapper/pkg/wrapper/nodeservice.go

```diff
@@ -125,10 +126,17 @@ func (s *NodeService) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 		podUid, volumeName := s.getPodUIDandVolumeName(targetPath)
 		savedPeerpodvolume.Labels["podUid"] = podUid
 		savedPeerpodvolume.Spec.PodUid = podUid
-		if volumeName != savedPeerpodvolume.Labels["volumeName"] {
-			glog.Error("The volume name from target path doesn't match with the CRD")
-			return
+		savedVolumeName := savedPeerpodvolume.Spec.VolumeName
```

Comment:
I think this is OK, but I am wondering why `.Spec.VolumeName` is used instead of `.Labels["volumeName"]`. Is there any specific reason for this change, or just for conciseness?

Reply:
This was done for conciseness. It felt a bit weird to me to rely on a label, when we have the same information already available as part of the CR spec